### ISSUE TITLE
Build error while trying to compile with CUDA support in VS2022

### ISSUE THREAD

**Issue by 0xFFD** [opencv/opencv#24464]
### System Information

OpenCV python version: 4.5.2
Operating System / Platform: Windows 10 22H2
Python version: 3.12

### Detailed description

CMake Configure Output

```
General configuration for OpenCV 4.5.2 =====================================
  Version control:               unknown

  Extra modules:
    Location (extra):            C:/Users/penguin/opencv/opencv_contrib-4.5.2/modules
    Version control (extra):     unknown

  Platform:
    Timestamp:                   2023-10-27T18:47:19Z
    Host:                        Windows 10.0.19045 AMD64
    CMake:                       3.27.7
    CMake generator:             Visual Studio 17 2022
    CMake build tool:            C:/Program Files/Microsoft Visual Studio/2022/Professional/MSBuild/Current/Bin/amd64/MSBuild.exe
    MSVC:                        1937
    Configuration:               Release

  CPU/HW features:
    Baseline:                    SSE SSE2 SSE3
      requested:                 SSE3
    Dispatched code generation:  SSE4_1 SSE4_2 FP16 AVX AVX2 AVX512_SKX
      requested:                 SSE4_1 SSE4_2 AVX FP16 AVX2 AVX512_SKX
      SSE4_1 (17 files):         + SSSE3 SSE4_1
      SSE4_2 (2 files):          + SSSE3 SSE4_1 POPCNT SSE4_2
      FP16 (1 files):            + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 AVX
      AVX (5 files):             + SSSE3 SSE4_1 POPCNT SSE4_2 AVX
      AVX2 (31 files):           + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2
      AVX512_SKX (7 files):      + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2 AVX_512F AVX512_COMMON AVX512_SKX

  C/C++:
    Built as dynamic libs?:      YES
    C++ standard:                11
    C++ Compiler:                C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe  (ver 19.37.32825.0)
    C++ flags (Release):         /DWIN32 /D_WINDOWS /W4 /GR  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:fast     /EHa /wd4127 /wd4251 /wd4324 /wd4275 /wd4512 /wd4589 /MP  /MD /O2 /Ob2 /DNDEBUG 
    C++ flags (Debug):           /DWIN32 /D_WINDOWS /W4 /GR  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:fast     /EHa /wd4127 /wd4251 /wd4324 /wd4275 /wd4512 /wd4589 /MP  /MDd /Zi /Ob0 /Od /RTC1 
    C Compiler:                  C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe
    C flags (Release):           /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:fast     /MP   /MD /O2 /Ob2 /DNDEBUG 
    C flags (Debug):             /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:fast     /MP /MDd /Zi /Ob0 /Od /RTC1 
    Linker flags (Release):      /machine:x64  /INCREMENTAL:NO 
    Linker flags (Debug):        /machine:x64  /debug /INCREMENTAL 
    ccache:                      NO
    Precompiled headers:         NO
    Extra dependencies:          cudart_static.lib nppc.lib nppial.lib nppicc.lib nppicom.lib nppidei.lib nppif.lib nppig.lib nppim.lib nppist.lib nppisu.lib nppitc.lib npps.lib cublas.lib cudnn.lib cufft.lib -LIBPATH:C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/lib/x64 -LIBPATH:C:/cuda/lib/x64
    3rdparty dependencies:

  OpenCV modules:
    To be built:                 aruco bgsegm bioinspired calib3d ccalib core cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev datasets dnn dnn_objdetect dnn_superres dpm face features2d flann fuzzy gapi hfs highgui img_hash imgcodecs imgproc intensity_transform line_descriptor mcc ml objdetect optflow phase_unwrapping photo plot python3 quality rapid reg rgbd saliency shape stereo stitching structured_light superres surface_matching text tracking ts video videoio videostab wechat_qrcode world xfeatures2d ximgproc xobjdetect xphoto
    Disabled:                    -
    Disabled by dependency:      -
    Unavailable:                 alphamat cnn_3dobj cvv freetype hdf java julia matlab ovis python2 python2 sfm viz
    Applications:                tests perf_tests apps
    Documentation:               NO
    Non-free algorithms:         NO

  Windows RT support:            NO

  GUI: 
    Win32 UI:                    YES
    VTK support:                 NO

  Media I/O: 
    ZLib:                        build (ver 1.2.11)
    JPEG:                        build-libjpeg-turbo (ver 2.0.6-62)
    WEBP:                        build (ver encoder: 0x020f)
    PNG:                         build (ver 1.6.37)
    TIFF:                        build (ver 42 - 4.2.0)
    JPEG 2000:                   build (ver 2.4.0)
    OpenEXR:                     build (ver 2.3.0)
    HDR:                         YES
    SUNRASTER:                   YES
    PXM:                         YES
    PFM:                         YES

  Video I/O:
    DC1394:                      NO
    FFMPEG:                      YES (prebuilt binaries)
      avcodec:                   YES (58.91.100)
      avformat:                  YES (58.45.100)
      avutil:                    YES (56.51.100)
      swscale:                   YES (5.7.100)
      avresample:                YES (4.0.0)
    GStreamer:                   NO
    DirectShow:                  YES
    Media Foundation:            YES
      DXVA:                      YES

  Parallel framework:            Concurrency

  Trace:                         YES (with Intel ITT)

  Other third-party libraries:
    Intel IPP:                   2020.0.0 Gold [2020.0.0]
           at:                   C:/Users/penguin/opencv/build/3rdparty/ippicv/ippicv_win/icv
    Intel IPP IW:                sources (2020.0.0)
              at:                C:/Users/penguin/opencv/build/3rdparty/ippicv/ippicv_win/iw
    Lapack:                      NO
    Eigen:                       NO
    Custom HAL:                  NO
    Protobuf:                    build (3.5.1)

  NVIDIA CUDA:                   YES (ver 10.0, CUFFT CUBLAS FAST_MATH)
    NVIDIA GPU arch:             61
    NVIDIA PTX archs:

  cuDNN:                         YES (ver 7.6.5)

  OpenCL:                        YES (NVD3D11)
    Include path:                C:/Users/penguin/opencv/opencv-4.5.2/3rdparty/include/opencl/1.2
    Link libraries:              Dynamic load

  Python 3:
    Interpreter:                 C:/Users/penguin/AppData/Local/Programs/Python/Python312/python.exe (ver 3.12)
    Libraries:                   C:/Users/penguin/AppData/Local/Programs/Python/Python312/libs/python312.lib (ver 3.12.0)
    numpy:                       C:/Users/penguin/AppData/Local/Programs/Python/Python312/Lib/site-packages/numpy/core/include (ver )
    install path:                C:/Users/penguin/AppData/Local/Programs/Python/Python312/Lib/site-packages/cv2/python-3.12

  Python (for build):            C:/Users/penguin/AppData/Local/Programs/Python/Python312/python.exe

  Java:                          
    ant:                         NO
    JNI:                         NO
    Java wrappers:               NO
    Java tests:                  NO

  Install to:                    C:/Users/penguin/opencv/build/install
-----------------------------------------------------------------
```

And the error that keeps occurring while I'm trying to build (I even tried to reinstall windows) log was too long so I uploaded the log to gist.

https://gist.github.com/0xFFD/d561a0c398c9c485a890000124957536

### Steps to reproduce

Sorry for the long build log in VS2022 but I've tried everything VS2019 a different version of OpenCV different version of Python, cuda and cudnn is installed and in path I checked. Also I have extra modules path set in CMake as you can see. I'm out of options right now so I'll gladly follow any advice related to this issue.

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by cudawarped** [comment]
See my response on the [forum](https://forum.opencv.org/t/build-error-while-trying-to-compile-with-cuda-support-in-vs2022/15019/2)

**Comment by 0xFFD** [comment]
I've tried VS 2017 but now I get "ImportError: DLL load failed: The specified module could not be found." I tried everything on web but still couldn't managed to get it right.

**Comment by cudawarped** [comment]
If you have compiled a Release build of the latest version of OpenCV and you
1. didn't have any existing python installs of OpenCV including manual copies of the .pyd,
2. specified your python dependencies correctly including the directory where the OpenCV python bindings should be installed
> -DPYTHON3_PACKAGES_PATH=<PATH_TO_PYTHON_DIST>/Lib/site-packages/
3. and don't have any extra dependencies on anything other than CUDA like gstreamer etc.

then you should not be facing this error.

That said if you are you can use Process Monitor as descirbed in [ImportError: DLL load failed while importing cv2: The specified module could not be found](https://www.jamesbowley.co.uk/nbs/ImportError_dll_load_failed_while_importing_cv2.html) to track down the missing dll and add it to 
> <PATH_TO_PYTHON_DIST>/Lib/site-packages/cv2/config.py

**Comment by 0xFFD** [comment]
I've already wrote a long thanks message on OpenCV Forums but I'll write it down here too for those who experience this problem in the future, follow @cudawarped's tutorial and you'll get your answer. In my case problem was cudnn64_8.dll even tho I installed cuDNN and CUDA via NVIDIA’s official website. 

https://www.jamesbowley.co.uk/nbs/ImportError_dll_load_failed_while_importing_cv2.html

Follow this tutorial if you are getting problems too, in my case the fix is basically downloading cudnn64_8.dll from web and putting that .dll inside "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.3\bin" path. Then re-open the terminal and problem was gone for me, I tried `print(cv2.getBuildInformation())` and it was working fine. Thanks again to @cudawarped for this awesome tutorial.
